### PR TITLE
Add pre-commit hook following icefall convention

### PR DESCRIPTION
**Pull request by ezerhouni**

### CODE REVIEW

**Comment by ezerhouni** [comment]
@csukuangfj It is highly inspired by icefall

**Comment by csukuangfj** [comment]
Thanks! 

Could you also use
https://github.com/k2-fsa/icefall/blob/master/.github/workflows/style_check.yml
to let the CI check the Python code style?

**Comment by ezerhouni** [comment]
I will work on the 2 PR first thing tomorrow morning

**Comment by csukuangfj** [comment]
Thanks!

**Comment by ezerhouni** [comment]
@csukuangfj Just added the file, is there something else to add to the CI ?

File: .github/workflows/style_check.yml [hunk 1]
```diff
@@ -42,7 +43,22 @@ jobs:
         with:
           python-version: ${{ matrix.python-version }}
 
-      - name: Check style with cpplint
```
Comment (csukuangfj) [inline .github/workflows/style_check.yml:45]: Could you please keep the style check for the C++ code?

You can create a file, say `python_style_check.yml`, for checking the python code.

**Review by csukuangfj — APPROVED** [review]

**Comment by csukuangfj** [comment]
Thanks! Merging.

**Pull request merged**